index.js

```diff
@@ -33,15 +33,15 @@ function HyperdriveHttp (getArchive) {
     var key = archive
       ? encoding.encode(archive.key)
       : segs.shift()
-    var filename = segs.shift()
+    var filename = segs.join('/')
     var op = 'get'
 
     try {
       // check if we are serving archive at root
       key = key.replace(/\.changes$/, '')
-      encoding.decode(key)
+      encoding.encode(Buffer.from(key, 'hex'))
```

Comment:
I thought so, but its not checking if its the right length. This is just a hacky way to make sure its a Dat key (if there is no key then key = filename). 

You think there is a better way to see if that first segment is a key or file?